pkg/cmd/template/schema_test.go

```diff
@@ -783,6 +783,60 @@ dvs1.yml:
 		assertFails(t, filesToProcess, expectedErr, cmdOpts)
 	})
 
+	t.Run("when a data value of the wrong type used in @schema/default annotation", func(t *testing.T) {
+
+		schemaYAML := `#@data/values-schema
+---
+#@schema/default True
+int: 1
+#@schema/default 0
+str: ""
+#@schema/default "string"
+bool: False
+`
+		templateYAML := `#@ load("@ytt:data", "data")
+---
+foo: #@ data.values.int
+foo: #@ data.values.str
+foo: #@ data.values.bool
+`
+		expectedErr := `
+One or more data values were invalid
+====================================
+
+schema.yml:
+    |
+  4 | int: 1
+    |
+
+    = found: boolean
+    = expected: integer (by schema.yml:4)
+
```

Comment:
This could be confusing to the uninitiated that the effective value of the node could be overridden by a `@schema/default`.

The deeper we go into implementing Schema, the more and more valuable the ability to reference a _range_ of source lines that we could display here. I suspect that's a mighty undertaking.

Until then, we at least should mention the potentially missing factor:

```suggestion
schema.yml:
    |
  4 | int: 1
    |

    = found: boolean
    = expected: integer (by schema.yml:4)
    = hint: is the default value set using @schema/default?

```

or similar.

Reply:
Tried out this change, and it would apply to all mismatched type errors, not just ones that had the `@schema/default` annotation on them. I had to change a fair amount of tests and left in a TODO to remove this hint eventually. lmk if that's what you were expecting or not.